tools/ddrgen/src/scanners/dwarf/AixDwarfParser.cpp

```diff
@@ -218,14 +217,16 @@ parseArray(const string data, Dwarf_Die currentDie, Dwarf_Error *error)
 				/* If the type's typeID is built-in, leave it as is, otherwise, add the current file's ID to make it unique */
 				int ID = toInt(arrayInfo[3]);
 				if (ID > 0) {
-					ID = ID + _fileID;
+					ID = ID + _fileID + NUM_BUILT_IN_TYPES;
+				} else {
+					ID = ID * -1;
 				}
 				type->_udata = 0;
-				type->_refdata = 0;
+				type->_refdata = ID;
```

Comment:
As we discussed in person: I strongly do not recommend making the refdata numbers dependent on the IDs extracted from the AIX stabstrings. You should define your own numbering which is guaranteed to be unique. We have no guarantees for the ranges of the numbers extracted from stabstrings.

My suggestion was to create a static number. Every time you create a new Die, associate that Die with the static number and increment it. As such, the first Die would be associated with X, the next with X + 1, etc. Please refer to the OSX version of this code.

As you said to me already, these IDs based on the AIX stabstring info are not necessarily guaranteed to be unique. For example, ID 101 in file 1000 would be the same as ID 100 in file 1001. This is a poor design unless you can guarantee that (1) the ID numbers are documented to always be in a specific range for projects of all sizes, (2) the file ids are documented to always be in a specific range for projects of all sizes, (3) the file and type ids are always unique when added together for all projects, and (4) the ranges of these values will never change in future versions of AIX/stabstring info.

The pair of file id and type id (not added together) is the best we've got for determining the type of a variable in the stabstring info. However, since we cannot guarantee the range of these values, adding them together is not a good idea. The best way to futureproof it and guarantee unique numbering for refdata is to come up with **your own** numbering and **not use refdata numbering that relies on IDs from AIX**.